packages/cadl-ranch-specs/http/azure/resource-manager/models/resources/extension.tsp

```diff
@@ -0,0 +1,179 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using Azure.Core;
+using Azure.ResourceManager;
+using TypeSpec.OpenAPI;
+
+namespace Azure.ResourceManager.Models.Resources;
+
+@resource("customExtensionResources")
+model CustomExtensionResource is ExtensionResource<CustomExtensionResourceProperties> {
+  @key("customExtensionResourceName")
+  @path
+  @segment("customExtensionResources")
+  @doc("arm custom extension resource name name for path")
+  @pattern("^[A-Za-z0-9]([A-Za-z0-9-_.]{0,62}[A-Za-z0-9])?$")
+  name: string;
+}
+
+/** CustomExtensionResource properties */
+model CustomExtensionResourceProperties {
+  @doc("The description of the resource.")
+  description?: string;
+
+  /** The status of the last operation. */
+  @visibility("read")
+  provisioningState?: ProvisioningState;
+}
+
+@armResourceOperations
+interface CustomExtensionResources {
+  @scenario
+  @scenarioDoc("""
+    Resource GET operation.
+    Expected path: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.Models.Resources/customExtensionResources/extension
+    Expected query parameter: api-version=2023-12-01-preview
```

Comment:
is it an extension resource?

Reply:
@tadelesh
I think this is an  extension resource, do you have any better ideas?

Reply:
Extension resource is an extension of another resource: https://learn.microsoft.com/sr-latn-rs/azure/azure-resource-manager/management/extension-resource-types

@v-hongli1 See if we could modify the path to be the extension of the tracked resource, e.g.
`/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.Models.Resources/topLevelTrackedResources/top/customExtensionResources/extension`